### PR TITLE
Add build for AF_ALG

### PR DESCRIPTION
**Pull request by JacobBarthelmeh**
adds --enable-afalg build

### CODE REVIEW

File: wolfcrypt/src/hash.c [hunk 1]
```diff
@@ -840,6 +840,7 @@ int wc_HashFree(wc_HashAlg* hash, enum wc_HashType type)
             }
             wc_Sha256Free(sha256);
         }
+	wc_Sha256Free(sha256);
```
Comment (toddouska) [inline wolfcrypt/src/hash.c]: Why are 2 Sha256Frees needed 2 lines apart?

File: wolfcrypt/src/sha256.c [hunk 2]
```diff
@@ -2572,6 +2577,10 @@ SHA256_NOINLINE static int Transform_Sha256_AVX2_RORX_Len(wc_Sha256* sha256,
 
 #elif defined(WOLFSSL_IMX6_CAAM) && !defined(NO_IMX6_CAAM_HASH)
     /* functions defined in wolfcrypt/src/port/caam/caam_sha256.c */
+
+//#elif defined(WOLFSSL_AFALG_HASH)
```
Comment (toddouska) [inline wolfcrypt/src/sha256.c]: Let's remove c++ style comments

File: wolfcrypt/src/hash.c (hunk 1 above)
Comment (JacobBarthelmeh) [reply wolfcrypt/src/hash.c]: One was if Init succeeded and the other if it failed. It's wrong to need it if Init failed, am updating the wc_InitSha256 function to fix this.

File: wolfcrypt/src/sha256.c (hunk 2 above)
Comment (JacobBarthelmeh) [reply wolfcrypt/src/sha256.c]: thanks! fixing it

**Pull request merged**